docs: update azurerm backend docs (#34564)
hashicorp/terraform: website/docs/language/settings/backends/azurerm.mdx

~> **Terraform 1.1 and 1.2 supported a feature-flag to allow enabling/disabling the use of Microsoft Graph (and MSAL) rather than Azure Active Directory Graph (and ADAL) - however this flag has since been removed in Terraform 1.3. Microsoft Graph (and MSAL) are now enabled by default and Azure Active Directory Graph (and ADAL) can no longer be used.

## Authentication

The `azurerm` backend supports 3 methods of authenticating to the storage account:

- **Access Key** (default)
- **Entra ID**
- **SAS Token**

> The **Access Key** method can be used directly or in combination with an Entra ID principal (e.g. user, service principal or managed identity). By default when you use an Entra ID principal, it will use the Access Token of that principal to generate an Access Key for the storage account and use that Access Key to authenticate to the state file blob. To use an Access Key directly you must generate one for you state file blob and pass it to the backend config.

Reviewer: ```suggestion
The **Access Key** method can be used directly or in combination with an Entra ID principal (e.g. user, service principal or managed identity). By default when you use an Entra ID principal, it will use the Access Token of that principal to generate an Access Key for the storage account and use that Access Key to authenticate to the state file blob. To use an Access Key directly you must generate one for your state file blob and pass it to the backend config.
```

  }
}
```

Connect to Storage Account with Entra ID:

```hcl
terraform {
  backend "azurerm" {
    resource_group_name  = "rg-tfstate-001"  # Can be passed via `-backend-config=`"resource_group_name=<resource group name>"` in the `init` command.
    storage_account_name = "stotfstate001"  # Can be passed via `-backend-config=`"storage_account_name=<storage account name>"` in the `init` command.
    container_name       = "prod"  # Can be passed via `-backend-config=`"container_name=<container name>"` in the `init` command.
    key                  = "terraform.tfstate"  # Can be passed via `-backend-config=`"key=<blob key name>"` in the `init` command.
    use_msi              = true  # Can also be set via `USE_MSI` environment variable.
    subscription_id      = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_SUBSCRIPTION_ID` environment variable.
    tenant_id            = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_TENANT_ID` environment variable.
    use_azuread_auth     = true  # Can also be set via `USE_AZURE_AD` environment variable.
  }
}
```

### Backend: Entra ID Service Principal via Client Secret

> ~> **Warning!**: This method requires you to manage and a rotate a secret. Consider using OIDC as a more secure approach.

Reviewer: ```suggestion
~> **Warning!**: This method requires you to manage and rotate a secret. Consider using OIDC as a more secure approach.
```

    storage_account_name = "stotfstate001"  # Can be passed via `-backend-config=`"storage_account_name=<storage account name>"` in the `init` command.
    container_name       = "prod"  # Can be passed via `-backend-config=`"container_name=<container name>"` in the `init` command.
    key                  = "terraform.tfstate"  # Can be passed via `-backend-config=`"key=<blob key name>"` in the `init` command.
    client_id            = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_CLIENT_ID` environment variable.
    client_secret        = "************************************"  # Can also be set via `ARM_CLIENT_SECRET` environment variable.
    subscription_id      = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_SUBSCRIPTION_ID` environment variable.
    tenant_id            = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_TENANT_ID` environment variable.
    use_azuread_auth     = true  # Can also be set via `USE_AZURE_AD` environment variable.
  }
}
```

### Backend: Entra ID Service Principal via Client Certificate

> ~> **Warning!**: This method requires you to manage and a rotate a secret. Consider using OIDC as a more secure approach.

Reviewer: ```suggestion
~> **Warning!**: This method requires you to manage and rotate a secret. Consider using OIDC as a more secure approach.
```

    tenant_id                   = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_TENANT_ID` environment variable.
    use_azuread_auth            = true  # Can also be set via `USE_AZURE_AD` environment variable.
  }
}
```

### Backend: Access Key Direct

~> **Warning!**: This method requires you to manage and a rotate a secret. Consider using OIDC as a more secure approach.

```hcl
terraform {
  backend "azurerm" {
    resource_group_name  = "rg-tfstate-001"  # Can be passed via `-backend-config=`"resource_group_name=<resource group name>"` in the `init` command.
    storage_account_name = "stotfstate001"  # Can be passed via `-backend-config=`"storage_account_name=<storage account name>"` in the `init` command.
    container_name       = "prod"  # Can be passed via `-backend-config=`"container_name=<container name>"` in the `init` command.
    key                  = "terraform.tfstate"  # Can be passed via `-backend-config=`"key=<blob key name>"` in the `init` command.
    access_key           = "abcdefghijklmnopqrstuvwxyz0123456789..."  # Can also be set via `ARM_ACCESS_KEY` environment variable.
  }
}
```

### Backend: SAS Token

> ~> **Warning!**: This method requires you to manage and a rotate a secret. Consider using OIDC as a more secure approach.

Reviewer: ```suggestion
~> **Warning!**: This method requires you to manage and rotate a secret. Consider using OIDC as a more secure approach.
```

You can use an App Registration (Service Principal) or a User Assigned Managed Identity to configure federated credentials. You must supply the client id of the principal.

Connect to Storage Account with Access Key (default):

```hcl
data "terraform_remote_state" "foo" {
  backend = "azurerm"
  config = {
    resource_group_name  = "rg-tfstate-001"
    storage_account_name = "stotfstate001"
    container_name       = "prod"
    key                  = "terraform.tfstate"
    use_oidc             = true  # Can also be set via `USE_OIDC` environment variable.
    client_id            = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_CLIENT_ID` environment variable.
    subscription_id      = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_SUBSCRIPTION_ID` environment variable.
    tenant_id            = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_TENANT_ID` environment variable.
  }
}
```

Connect to Storage Account with Entra ID:

```hcl
data "terraform_remote_state" "foo" {
  backend = "azurerm"
  config = {

>     resource_group_name  = "rg-tfstate-001"
>     storage_account_name = "stotfstate001"
>     container_name       = "prod"
>     key                  = "terraform.tfstate"
>     use_oidc             = true  # Can also be set via `USE_OIDC` environment variable.
>     client_id            = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_CLIENT_ID` environment variable.
>     subscription_id      = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_SUBSCRIPTION_ID` environment variable.
>     tenant_id            = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_TENANT_ID` environment variable.
>     use_azuread_auth     = true  # Can also be set via `USE_AZURE_AD` environment variable.

Reviewer 2: Formatting, fix names, and also retain the original example values

```suggestion
    resource_group_name  = "StorageAccount-ResourceGroup"
    storage_account_name = "terraform123abc"
    container_name       = "tfstate"
    key                  = "prod.terraform.tfstate"
    use_oidc             = true                                    # Can also be set via `ARM_USE_OIDC` environment variable.
    client_id            = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_CLIENT_ID` environment variable.
    subscription_id      = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_SUBSCRIPTION_ID` environment variable.
    tenant_id            = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_TENANT_ID` environment variable.
    use_azuread_auth     = true                                    # Can also be set via `ARM_USE_AZUREAD` environment variable.
```

  }
}
```

> ### Data Source: Entra ID Managed Identity Principal

Reviewer 2: ```suggestion
### Data Source: Azure AD Managed Identity Principal
```

> You can use a User Assigned Managed Identity as well as a System Assigned Managed Identity on your agent / runner compute. However the authentication does not support specifiying the Client ID of the User Assigned Managed Identity, so you can only supply one per compute instance.

Reviewer 2: ```suggestion
You can use a User Assigned Managed Identity as well as a System Assigned Managed Identity on your agent / runner compute environment. However the backend does not currently support specifying the Client ID of the User Assigned Managed Identity, so you can only supply one per compute instance.
```

> Connect to Storage Account with Access Key (default):

Reviewer 2: ```suggestion
*Connect to Storage Account with Access Key*
```

```hcl
data "terraform_remote_state" "foo" {
  backend = "azurerm"
  config = {

>     resource_group_name  = "rg-tfstate-001"
>     storage_account_name = "stotfstate001"
>     container_name       = "prod"
>     key                  = "terraform.tfstate"
>     use_msi              = true  # Can also be set via `USE_MSI` environment variable.
>     subscription_id      = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_SUBSCRIPTION_ID` environment variable.
>     tenant_id            = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_TENANT_ID` environment variable.

Reviewer 2: Formatting, fix name, and also retain the original example values

```suggestion
    resource_group_name  = "StorageAccount-ResourceGroup"
    storage_account_name = "terraform123abc"
    container_name       = "tfstate"
    key                  = "prod.terraform.tfstate"
    use_msi              = true                                    # Can also be set via `ARM_USE_MSI` environment variable.
    subscription_id      = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_SUBSCRIPTION_ID` environment variable.
    tenant_id            = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_TENANT_ID` environment variable.
```

  }
}
```

> Connect to Storage Account with Entra ID:

Reviewer 2: ```suggestion
*Connect to Storage Account with Azure Active Directory authentication*
```

```hcl
data "terraform_remote_state" "foo" {
  backend = "azurerm"
  config = {

>     resource_group_name  = "rg-tfstate-001"
>     storage_account_name = "stotfstate001"
>     container_name       = "prod"
>     key                  = "terraform.tfstate"
>     use_msi              = true  # Can also be set via `USE_MSI` environment variable.
>     subscription_id      = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_SUBSCRIPTION_ID` environment variable.
>     tenant_id            = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_TENANT_ID` environment variable.
>     use_azuread_auth     = true  # Can also be set via `USE_AZURE_AD` environment variable.

Reviewer 2: Formatting, fix name, and also retain the original example values

```suggestion
    resource_group_name  = "StorageAccount-ResourceGroup"
    storage_account_name = "terraform123abc"
    container_name       = "tfstate"
    key                  = "prod.terraform.tfstate"
    use_msi              = true                                    # Can also be set via `ARM_USE_MSI` environment variable.
    subscription_id      = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_SUBSCRIPTION_ID` environment variable.
    tenant_id            = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_TENANT_ID` environment variable.
    use_azuread_auth     = true                                    # Can also be set via `ARM_USE_AZUREAD` environment variable.
```

  }
}
```

> ### Data Source: Entra ID Service Principal via Client Secret

Reviewer 2: ```suggestion
### Data Source: Azure AD Service Principal via Client Secret
```

> ~> **Warning!**: This method requires you to manage and a rotate a secret. Consider using OIDC as a more secure approach.

Reviewer: ```suggestion
~> **Warning!**: This method requires you to manage and rotate a secret. Consider using OIDC as a more secure approach.
```

> Connect to Storage Account with Access Key (default):

Reviewer 2: ```suggestion
*Connect to Storage Account with Access Key*
```

```hcl
data "terraform_remote_state" "foo" {
  backend = "azurerm"
  config = {

>     resource_group_name  = "rg-tfstate-001"
>     storage_account_name = "stotfstate001"
>     container_name       = "prod"
>     key                  = "terraform.tfstate"
>     client_id            = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_CLIENT_ID` environment variable.
>     client_secret        = "************************************"  # Can also be set via `ARM_CLIENT_SECRET` environment variable.
>     subscription_id      = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_SUBSCRIPTION_ID` environment variable.
>     tenant_id            = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_TENANT_ID` environment variable.

Reviewer 2: Retain the original example values

```suggestion
    resource_group_name  = "StorageAccount-ResourceGroup"
    storage_account_name = "terraform123abc"
    container_name       = "tfstate"
    key                  = "prod.terraform.tfstate"
    client_id            = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_CLIENT_ID` environment variable.
    client_secret        = "************************************"  # Can also be set via `ARM_CLIENT_SECRET` environment variable.
    subscription_id      = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_SUBSCRIPTION_ID` environment variable.
    tenant_id            = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_TENANT_ID` environment variable.
```

  }
}
```

> Connect to Storage Account with Entra ID:

Reviewer 2: ```suggestion
*Connect to Storage Account with Azure Active Directory authentication*
```

```hcl
data "terraform_remote_state" "foo" {
  backend = "azurerm"
  config = {

>     resource_group_name  = "rg-tfstate-001"
>     storage_account_name = "stotfstate001"
>     container_name       = "prod"
>     key                  = "terraform.tfstate"
>     client_id            = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_CLIENT_ID` environment variable.
>     client_secret        = "************************************"  # Can also be set via `ARM_CLIENT_SECRET` environment variable.
>     subscription_id      = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_SUBSCRIPTION_ID` environment variable.
>     tenant_id            = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_TENANT_ID` environment variable.
>     use_azuread_auth     = true  # Can also be set via `USE_AZURE_AD` environment variable.

Reviewer 2: Formatting, fix name, and also retain the original example values

```suggestion
    resource_group_name  = "StorageAccount-ResourceGroup"
    storage_account_name = "terraform123abc"
    container_name       = "tfstate"
    key                  = "prod.terraform.tfstate"
    client_id            = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_CLIENT_ID` environment variable.
    client_secret        = "************************************"  # Can also be set via `ARM_CLIENT_SECRET` environment variable.
    subscription_id      = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_SUBSCRIPTION_ID` environment variable.
    tenant_id            = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_TENANT_ID` environment variable.
    use_azuread_auth     = true                                    # Can also be set via `ARM_USE_AZUREAD` environment variable.
```

  }
}
```

> ### Data Source: Entra ID Service Principal via Client Certificate

Reviewer 2: ```suggestion
### Data Source: Azure AD Service Principal via Client Certificate
```

> ~> **Warning!**: This method requires you to manage and a rotate a secret. Consider using OIDC as a more secure approach.

Reviewer: ```suggestion
~> **Warning!**: This method requires you to manage and rotate a secret. Consider using OIDC as a more secure approach.
```

> Connect to Storage Account with Access Key (default):

Reviewer 2: ```suggestion
*Connect to Storage Account with Access Key*
```

```hcl
data "terraform_remote_state" "foo" {
  backend = "azurerm"
  config = {

>     resource_group_name  = "rg-tfstate-001"
>     storage_account_name = "stotfstate001"
>     container_name       = "prod"
>     key                  = "terraform.tfstate"
>     client_id                   = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_CLIENT_ID` environment variable.
>     client_certificate_path     = "./cert.pfx"  # Can also be set via `ARM_CLIENT_CERTIFICATE_PATH` environment variable.
>     client_certificate_password = "************************************"  # Can also be set via `ARM_CLIENT_CERTIFICATE_PASSWORD` environment variable.
>     subscription_id             = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_SUBSCRIPTION_ID` environment variable.
>     tenant_id                   = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_TENANT_ID` environment variable.

Reviewer 2: Formatting, and also retain the original example values

```suggestion
    resource_group_name         = "StorageAccount-ResourceGroup"
    storage_account_name        = "terraform123abc"
    container_name              = "tfstate"
    key                         = "prod.terraform.tfstate"
    client_id                   = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_CLIENT_ID` environment variable.
    client_certificate_path     = "/path/to/bundle.pfx"                   # Can also be set via `ARM_CLIENT_CERTIFICATE_PATH` environment variable.
    client_certificate_password = "************************************"  # Can also be set via `ARM_CLIENT_CERTIFICATE_PASSWORD` environment variable.
    subscription_id             = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_SUBSCRIPTION_ID` environment variable.
    tenant_id                   = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_TENANT_ID` environment variable.
```

  }
}
```

> Connect to Storage Account with Entra ID:

Reviewer 2: ```suggestion
*Connect to Storage Account with Azure Active Directory authentication*
```

```hcl
data "terraform_remote_state" "foo" {
  backend = "azurerm"
  config = {

>     resource_group_name  = "rg-tfstate-001"
>     storage_account_name = "stotfstate001"
>     container_name       = "prod"
>     key                  = "terraform.tfstate"
>     client_id                   = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_CLIENT_ID` environment variable.
>     client_certificate_path     = "./cert.pfx"  # Can also be set via `ARM_CLIENT_CERTIFICATE_PATH` environment variable.
>     client_certificate_password = "************************************"  # Can also be set via `ARM_CLIENT_CERTIFICATE_PASSWORD` environment variable.
>     subscription_id             = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_SUBSCRIPTION_ID` environment variable.
>     tenant_id                   = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_TENANT_ID` environment variable.
>     use_azuread_auth            = true  # Can also be set via `USE_AZURE_AD` environment variable.

Reviewer 2: Formatting, and also retain the original example values

```suggestion
    resource_group_name         = "StorageAccount-ResourceGroup"
    storage_account_name        = "terraform123abc"
    container_name              = "tfstate"
    key                         = "prod.terraform.tfstate"
    client_id                   = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_CLIENT_ID` environment variable.
    client_certificate_path     = "/path/to/bundle.pfx"                   # Can also be set via `ARM_CLIENT_CERTIFICATE_PATH` environment variable.
    client_certificate_password = "************************************"  # Can also be set via `ARM_CLIENT_CERTIFICATE_PASSWORD` environment variable.
    subscription_id             = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_SUBSCRIPTION_ID` environment variable.
    tenant_id                   = "00000000-0000-0000-0000-000000000000"  # Can also be set via `ARM_TENANT_ID` environment variable.
    use_azuread_auth            = true                                    # Can also be set via `ARM_USE_AZUREAD` environment variable.
```

  }
}
```

### Data Source: Access Key Direct

> ~> **Warning!**: This method requires you to manage and a rotate a secret. Consider using OIDC as a more secure approach.

Reviewer: ```suggestion
~> **Warning!**: This method requires you to manage and rotate a secret. Consider using OIDC as a more secure approach.
```

```hcl
data "terraform_remote_state" "foo" {
  backend = "azurerm"
  config = {

>     resource_group_name  = "rg-tfstate-001"
>     storage_account_name = "stotfstate001"
>     container_name       = "prod"
>     key                  = "terraform.tfstate"
>     access_key           = "abcdefghijklmnopqrstuvwxyz0123456789..."  # Can also be set via `ARM_ACCESS_KEY` environment variable.

Reviewer 2: Retain the original example values

```suggestion
    resource_group_name  = "StorageAccount-ResourceGroup"
    storage_account_name = "terraform123abc"
    container_name       = "tfstate"
    key                  = "prod.terraform.tfstate"
    access_key           = "abcdefghijklmnopqrstuvwxyz0123456789..."  # Can also be set via `ARM_ACCESS_KEY` environment variable.
```

  }
}
```

### Data Source: SAS Token

> ~> **Warning!**: This method requires you to manage and a rotate a secret. Consider using OIDC as a more secure approach.

Reviewer: ```suggestion
~> **Warning!**: This method requires you to manage and rotate a secret. Consider using OIDC as a more secure approach.
```

```hcl
data "terraform_remote_state" "foo" {
  backend = "azurerm"
  config = {

>     resource_group_name  = "rg-tfstate-001"
>     storage_account_name = "stotfstate001"
>     container_name       = "prod"
>     key                  = "terraform.tfstate"
>     sas_token            = "abcdefghijklmnopqrstuvwxyz0123456789..."  # Can also be set via `ARM_SAS_TOKEN` environment variable.

Reviewer 2: Retain the original example values

```suggestion
    resource_group_name  = "StorageAccount-ResourceGroup"
    storage_account_name = "terraform123abc"
    container_name       = "tfstate"
    key                  = "prod.terraform.tfstate"
    sas_token            = "abcdefghijklmnopqrstuvwxyz0123456789..."  # Can also be set via `ARM_SAS_TOKEN` environment variable.
```

  }
}
```

## Configuration Variables

> !> ****Warning!**:**  We recommend using environment variables to supply credentials and other sensitive data. If you use `-backend-config` or hardcode these values directly in your configuration, Terraform will include these values in both the `.terraform` subdirectory and in plan files. Refer to [Credentials and Sensitive Data](/terraform/language/settings/backends/configuration#credentials-and-sensitive-data) for details.

Reviewer 2: This already renders correctly?

```suggestion
!> **Warning:**  We recommend using environment variables to supply credentials and other sensitive data. If you use `-backend-config` or hardcode these values directly in your configuration, Terraform will include these values in both the `.terraform` subdirectory and in plan files. Refer to [Credentials and Sensitive Data](/terraform/language/settings/backends/configuration#credentials-and-sensitive-data) for details.
```

***

> When authenticating using a Service Principal with OpenID Connect (OIDC / Workload identity federation)  - the following fields are also supported:

Reviewer 2: ```suggestion
When authenticating using a Service Principal with OpenID Connect (OIDC / Workload Identity Federation) - the following fields are also supported:
```

> When authenticating using a Service Principal with OpenID Connect (OIDC / Workload identity federation)  - the following fields are also supported:

Reviewer 2: ```suggestion
When authenticating using a Service Principal with OpenID Connect (OIDC / Workload Identity Federation) - the following fields are also supported:
```

***

> When authenticating using Entra ID service principal, you have the option to use Entra ID authentication for the storage account (as opposed to Access Key or SAS Token) - the following fields are also supported:

Reviewer 2: ```suggestion
When authenticating using an Azure AD Service Principal, you have the option to use Azure Active Directory authentication for the storage account (rather than by an Access Key or SAS Token) - the following fields are also supported:
```

> * `use_azuread_auth` - (Optional) Should EntraID Authentication be used to access the Blob Storage Account. This can also be sourced from the `ARM_USE_AZUREAD` environment variable.

Reviewer 2: ```suggestion
* `use_azuread_auth` - (Optional) Whether Azure Active Directory Authentication should be used to access the Blob Storage Account. This can also be sourced from the `ARM_USE_AZUREAD` environment variable.
```

> -> **Note:** When using Entra ID for Authentication to Storage you need to ensure the `Storage Blob Data Owner` or `Container Blob Data Owner` roles are assigned.

Reviewer 2: ```suggestion
-> **Note:** When using Azure Active Directory Authentication, you must ensure the `Storage Blob Data Owner` or `Container Blob Data Owner` roles are assigned to your Storage Account.
```
